tests/pytests/test_empty.py

```diff
@@ -0,0 +1,655 @@
+    env.assertEqual(int(res[0]), 500)
+
+def testEmptyText():
+    """Tests the indexing and querying of empty TEXT (field type) values"""
+
+    env = Env(moduleArgs="DEFAULT_DIALECT 2")
+    conn = getConnectionByEnv(env)
+
+    def testEmptyTextHash(idx):
+        """Tests the indexing and querying of empty values for a TEXT field of a
+        hash index
+        Extensive tests are added here, specifically to the query part, due to
+        the addition of the `isempty` function syntax added to the parser.
+        """
+
+        # Populate the db with a document that has an empty value for a TEXT field
+        conn.execute_command('HSET', 'h1', 't', '')
+
+        # ------------------------- Simple retrieval ---------------------------
+        # Search for a single document, via its indexed empty value
+        cmd = f'FT.SEARCH {idx} isempty(@t)'.split(' ')
+        expected = [1, 'h1', ['t', '']]
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------ Negation ------------------------------
+        # Search for a negation of an empty value, make sure the document is NOT
+        # returned
+        cmd = f'FT.SEARCH {idx} -isempty(@t)'.split(' ')
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # Search for a negation of a non-empty value, make sure the document is
+        # returned
+        cmd = f'FT.SEARCH {idx} -@t:foo'.split(' ')
+        expected = [1, 'h1', ['t', '']]
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------- Union --------------------------------
+        # Union of empty and non-empty values
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['isempty(@t) | @t:foo']
+        expected = [1, 'h1', ['t', '']]
+        cmd_assert(env, cmd, expected)
+
+        # Same in opposite order
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['@t:foo | isempty(@t)']
+        cmd_assert(env, cmd, expected)
+
+        # ---------------------------- Intersection ----------------------------
+        # Empty intersection
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['isempty(@t) @t:foo']
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # Non-empty intersection
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['isempty(@t) -@t:foo']
+        expected = [1, 'h1', ['t', '']]
+        cmd_assert(env, cmd, expected)
+
+        # Same in opposite order
+        cmd = f'FT.SEARCH {idx}'.split(' ') + ['-@t:foo isempty(@t)']
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------- Prefix -------------------------------
+        cmd = f'FT.SEARCH {idx} @t:*pty'.split(' ')
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
+
+        # ------------------------------- Suffix -------------------------------
+        cmd = f'FT.SEARCH {idx} @t:__em*'.split(' ')
+        expected = EMPTY_RESULT
+        cmd_assert(env, cmd, expected)
```

Comment:
Irrelevant now when using `isempty(@t)` instead of `__empty`?

Reply:
A little bit. But not too much since it doesn't hurt to show that this prefix search doesn't yield empty results.